File: core/EncodeAndDecodeTinyurl/EncodeAndDecodeTinyurl.java
package com.dbc.code;

import java.util.HashMap;
import java.util.Map;

public class EncodeAndDecodeTinyurl {
    private final Map<String, String> urlMap = new HashMap<>();
    private final Map<String, String> tinyUrlMap = new HashMap<>();
    private int count = 0;

    public String encode(String longUrl) {
        if (!this.urlMap.containsKey(longUrl)) {
            this.count++;
            String temp = "XHID62" + this.count;
            this.urlMap.put(longUrl, temp);
            this.tinyUrlMap.put(temp, longUrl);
        }
        return this.urlMap.get(longUrl);
    }

    public String decode(String shortUrl) {
        return this.tinyUrlMap.get(shortUrl);
    }
}

File: core/EncodeAndDecodeTinyurl/EncodeAndDecodeTinyurl.py
import uuid

class Codec:
    def __init__(self):
        self.url_map = {}
        self.tinyurl_map = {}

    def encode(self, longUrl: str) -> str:
        if longUrl not in self.url_map:
            temp = str(uuid.uuid1()).replace("-", "")
            self.url_map[longUrl] = temp
            self.tinyurl_map[temp] = longUrl

        return f"https://tinyurl.com/{self.url_map[longUrl]}"

    def decode(self, shortUrl: str) -> str:
        return self.tinyurl_map[shortUrl.split("/")[-1]]








